_std/component_defines.dm

```diff
@@ -82,11 +82,58 @@
 //When a mob dies
 #define COMSIG_MOB_DEATH "mob_death"
 
+//attack_X signals
+/// Attacking wiht an item in-hand
+#define COMSIG_ATTACKBY "attackby"
+/// Bitflag return of an attackby proc successfuly reacting (based on it's own conditions)
+/// I dunno, feel free to improve this; make global bitflag returns for all components.
+#define COMSIGBIT_ATTACKBY_COMPLETE 1
+
 
 // projectile signals
 /// After a projectile makes a valid hit on an atom (after immunity/other early returns, before other effects)
 #define COMSIG_PROJ_COLLIDE "proj_collide_atom"
 
+// MechComp signals - Content signals - Use these in you MechComp compatible devices
+/// Add an input chanel for a device to send into
+#define COMSIG_MECHCOMP_ADD_INPUT "mechcomp_add_input"
+/// Connect two mechcomp devices together
+#define COMSIG_MECHCOMP_ADD_CONFIG "mechcomp_add_config"
+/// Connect two mechcomp devices together
+#define COMSIG_MECHCOMP_ALLOW_MANUAL_SIGNAL "mechcomp_allow_manual_sigset"
+/// Remove all connected devices     //MarkNstein needs attention
```

Comment:
X

Reply:
Ack, forgot to to "Find in Folder" the **whole** project. Thanks